### PR TITLE
doc: versioning policy and feature stages docs. Part of #4533

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
versioning policy and feature status docs
Part of #4533

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: docs/release/feature-stages.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+* Refer to the Kubeflow Pipelines [introduction]((https://www.kubeflow.org/docs/pipelines/overview/)) for
+  - An [overview](https://www.kubeflow.org/docs/pipelines/overview/pipelines-overview/) and [main concepts](https://www.kubeflow.org/docs/pipelines/overview/concepts/), for example, what is a pipeline or run in Kubeflow Pipelines.
+  - How to manage the Kubeflow Pipelines resources via the the [Kubeflow Pipelines UI](https://www.kubeflow.org/docs/pipelines/overview/interfaces/).
+* Refer to the Kubeflow Pipelines [SDK samples and tutorials](https://www.kubeflow.org/docs/pipelines/tutorials/sdk-examples/)
+for how to manage the Kubeflow Pipelines resources via the Kubeflow Pipelines client.
+* A complete API method list and a detailed description of parameters to those API methods are available at the Kubeflow Pipelines [API reference](https://www.kubeflow.org/docs/pipelines/reference/api/kubeflow-pipeline-api-spec/).
+
+## Features in Beta
+
+* Job API
```
Comment (joeliedtke) [inline docs/release/feature-stages.md:51]: Consider adding a description or link that describes what the Job API does.

File: docs/release/feature-stages.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  |                  | Upload | Create | Get | List | Delete | Archive | Unarchive | Enable | Disable | Terminate | Retry |
+  |:----------------:|:------:|:------:|:---:|:----:|:------:|:-------:|:---------:|:------:|:-------:|:---------:|:-----:|
+  | Job              |        |  ☑️ |☑️|☑️ | ☑️  |         |           | ☑️  | ☑️   |
+
+
+* [Upgrade support for the Kubeflow Pipelines standalone deployment](https://www.kubeflow.org/docs/pipelines/installation/standalone-deployment/#upgrading-kubeflow-pipelines).
+
+* [Built-in Visualizations](https://www.kubeflow.org/docs/pipelines/sdk/output-viewer/).
```
Comment (joeliedtke) [inline docs/release/feature-stages.md:60]: For the next three list items, please revise to use sentence case.

Built-in visualizations
Pipeline metrics
Multi-user isolation

File: docs/release/feature-stages.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  `client.pipeline_uploads.upload_pipeline()`.
+  
+  Refer to [Using the Kubeflow Pipelines SDK](https://www.kubeflow.org/docs/pipelines/tutorials/sdk-examples/) for some example usages and [SDK generated API reference](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html#generated-apis) for exact method signatures.
+
+* [Step caching](https://www.kubeflow.org/docs/pipelines/caching/).
+
+* [ML Metadata (MLMD)](https://github.com/google/ml-metadata) UI integration.
+  For example, artifact and execution list and detail pages.
+
+* [Python based custom visualization](https://www.kubeflow.org/docs/pipelines/sdk/python-based-visualizations/).
```
Comment (joeliedtke) [inline docs/release/feature-stages.md:83]: visualization -> visualizations

File: docs/release/feature-stages.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+* [Built-in Visualizations](https://www.kubeflow.org/docs/pipelines/sdk/output-viewer/).
+
+* [Pipeline Metrics](https://www.kubeflow.org/docs/pipelines/sdk/pipelines-metrics/).
+
+* [Multi-user Isolation](https://www.kubeflow.org/docs/pipelines/multi-user/).
+
+## Features in Alpha
+
+* [SDK client helper methods](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html). The helper methods are mainly maintained by
```
Comment (joeliedtke) [inline docs/release/feature-stages.md:68]: Question: Are all of the methods on the kfp.Client page included in Alpha, or is it specific methods? If it is specific methods, how do users determine which are alpha?

Suggested revision:

* [SDK client helper methods](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html). The helper methods are mainly maintained by the community. These methods can be convenient to use, but they may not provide all the features from the API and they may lack testing. These issues prevent us from moving these methods to Beta stage at this time. For more information, refer to the [SDK client Beta blockers project](https://github.com/kubeflow/pipelines/projects/7).

  We recommend and support the auto-generated client APIs instead. For example,
  `client.pipelines.list_pipelines()`, `client.runs.list_runs()` and
  `client.pipeline_uploads.upload_pipeline()`.

  For more information, refer to the [Using the Kubeflow Pipelines SDK](https://www.kubeflow.org/docs/pipelines/tutorials/sdk-examples/) guide for examples of using the SDK, and the [SDK  API reference](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html#generated-apis).

File: docs/release/versioning-policy.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Kubeflow Pipelines Versioning Policy
+
+The Kubeflow Pipelines version format follows [the semantic versioning](https://semver.org/).
```
Comment (joeliedtke) [inline docs/release/versioning-policy.md:3]: Suggestion revision:

The Kubeflow Pipelines version format follows [semantic versioning](https://semver.org/).

or

The Kubeflow Pipelines version format follows the [semantic versioning specification](https://semver.org/).

File: docs/release/versioning-policy.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+The Kubeflow Pipelines versions are in the format of `X.Y.Z`, where `X` is the major version,
```
Comment (joeliedtke) [inline docs/release/versioning-policy.md:4]: Remove the leading "The"

File: docs/release/versioning-policy.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+`Y` is the minor version, and `Z` is the patch version.
+
+As a guideline, we increment the:
+* MAJOR version when we make breaking changes on [stable features](./feature-status.md#stable-features),
```
Comment (joeliedtke) [inline docs/release/versioning-policy.md:8]: Replace the trailing "," with a "."

File: docs/release/versioning-policy.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+* MINOR version when we add functionality in a backwards compatible manner, and
```
Comment (joeliedtke) [inline docs/release/versioning-policy.md:9]: , and -> .

File: docs/release/versioning-policy.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+* PATCH version when we make backwards compatible bug fixes.
+
+In reality, we are not following the guideline strictly, e.g. PATCH versions might have backward compatible new functionalities that is low-risk.
```
Comment (joeliedtke) [inline docs/release/versioning-policy.md:12]: Suggested revision:

In reality, we are not following the guideline strictly. For example, a PATCH version might include a backward compatible new feature that is considered to be low-risk.

File: docs/release/versioning-policy.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+Additionally, we do pre-releases as an extension in the format of `X.Y.Z-rc.N`
+where `N` is a number. The appendix indicates the Nth release candidate before
+an upcoming public release named `X.Y.Z`.
+
+The Kubeflow Pipelines version `X.Y.Z` refers to the version (git tag) of the released
+Kubeflow Pipelines. It versions all released artifacts, including:
```
Comment (joeliedtke) [inline docs/release/versioning-policy.md:19]: Suggested revision:

It versions all released artifacts, including:

->

Versions include all released artifacts, such as:

File: docs/release/versioning-policy.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+* `kfp` and `kfp-server-api` python packages
```
Comment (joeliedtke) [inline docs/release/versioning-policy.md:20]: Please update the list items to be sentence case and end with a period.

* `kfp` and `kfp-server-api` python packages.
* Install manifests.
* Docker images on gcr.io/ml-pipeline.
* First party components.

File: docs/release/feature-stages.md (hunk 4 above)
Comment (Bobgy) [reply docs/release/feature-stages.md:68]: Thanks! I've added stable methods in the stable section. Detailed breakdown of each method's stage level should be added in the reference doc: https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html.
Work item planned in https://github.com/kubeflow/pipelines/issues/4533.

File: docs/release/versioning-policy.md (hunk 11 above)
Comment (Bobgy) [reply docs/release/versioning-policy.md:20]: Thank you!
When will it be suitable to list items in non-sentence casing?

**Comment by Bobgy** [comment]
Hi @joeliedtke, thanks a lot for the really helpful feedbacks.
I've updated accordingly.

**Comment by Bobgy** [comment]
Introduced in today's community meeting, I'll merge this.
/approve

**Pull request merged**